source/adapters/level_zero/v2/command_buffer.cpp

```diff
@@ -0,0 +1,309 @@
+//===--------- command_buffer.cpp - Level Zero Adapter ---------------===//
+//
+// Copyright (C) 2024 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "command_buffer.hpp"
+#include "../helpers/kernel_helpers.hpp"
+#include "logger/ur_logger.hpp"
+#include "../ur_interface_loader.hpp"
+
+namespace {
+
+// Checks whether zeCommandListImmediateAppendCommandListsExp can be used for a
+// given Context and Device.
+void checkImmediateAppendSupport(ur_context_handle_t Context,
+                                 ur_device_handle_t Device) {
+  // TODO The L0 driver is not reporting this extension yet. Once it does,
+  // switch to using the variable zeDriverImmediateCommandListAppendFound.
+
+  // Minimum version that supports zeCommandListImmediateAppendCommandListsExp.
+  constexpr uint32_t MinDriverVersion = 30898;
+  bool DriverSupportsImmediateAppend =
+      Context->getPlatform()->isDriverVersionNewerOrSimilar(1, 3,
```

Comment:
This is out of sync from the changes in https://github.com/oneapi-src/unified-runtime/pull/2498 could we share this `checkImmediateAppendSupport` helper between v1 and v2 adapters to avoid duplicating code and it becoming inconsistent.

Reply:
That makes sense, if v2 unconditionally uses immediate command-lists then we don't need to be so conservative. The `isPVC` check was also there, because that's the platform where we've verified this path for v1, but again for v2 purposes I don't think we need to be as conservative.